test/functional/tool_wallet.py

```diff
@@ -59,12 +101,32 @@ def run_test(self):
             Address Book: 3
         ''')
         self.assert_tool_output(out, '-wallet=wallet.dat', 'info')
-
-        # mutate the wallet to check the info command output changes accordingly
+        shasum_after = self.wallet_shasum()
+        timestamp_after = self.wallet_timestamp(' after info')
+        self.log.info('Setting wallet file permissions to read/write')
+        os.chmod(self.wallet_path(), stat.S_IRUSR | stat.S_IWUSR) # Set wallet permissions to 600
+        self.log_wallet_permissions()
+        assert_equal(shasum_before, shasum_after)
+        self.log.info('Wallet file shasum unchanged')
+        self.log_wallet_timestamp_comparison(timestamp_before, timestamp_after)
+        #
+        # TODO: Wallet tool info should not write to the wallet file.
+        # This assertion should be uncommented and succeed:
+        # assert_equal(timestamp_before, timestamp_after)
+
+        # Mutate the wallet to verify that the info command output changes accordingly.
+        self.log.info('Starting node')
         self.start_node(0)
+
+        self.log.info('Generating transaction to mutate wallet')
         self.nodes[0].generate(1)
+
+        self.log.info('Stopping node\n')
         self.stop_node(0)
 
+        self.log.info('Calling wallet tool info after generating a transaction')
+        shasum_before = self.wallet_shasum()
```

Comment:
Looks like you are overwriting a previous symbol with the same name. I think it is clearer to split the tests into smaller ones, so that overwriting does not happen. I.e.

```py
  self run_test(self):
   self.run_test_foo_bar()
   self.run_test_gen_tx()

  def run_test_foo_bar(self):
    assert True

...

Reply:
Done.